### PR TITLE
add new 'inbound' integration

### LINKED ISSUE #15: Support short form and unified format in one integration
Select the correct integration to use based on the format of the submitted lead data. This is high priority.

### PR DESCRIPTION
**Pull request by cgrayson**
that can take either 'shortform' or 'unified' lead XML. Closes #15

### CODE REVIEW

File: src/inbound.coffee [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# Response Function ------------------------------------------------------
+#
+
+response = (req, vars) ->
+  # response for both formats is the same
+  unified.response req, vars
+
+
+response.variables = ->
+  unified.response.variables
```
Comment (alexkwolfe) [inline src/inbound.coffee]: This is returning the `unified.response.variables` function, where I think you want to return the invocation of that function.

**Comment by cgrayson** [comment]
@alexkwolfe I changed this as we discussed, see especially the last commit, let me know what you think, please. (No rush, it's not going out late on a Friday anyway. :wink: )

**Comment by cgrayson** [comment]
@alexkwolfe review when you have a moment, please?

**Comment by alexkwolfe** [comment]
I see there is a `lendingtree.product` field in the request variables. I think that's meant to be populated (with some unknown values) so that the user can distinguish between different kinds of leads. I don't know what the different product names are. Maybe @kcaponeAP knows more about this.

**Comment by zalary** [comment]
iirc product tells you whether it's longform vs shortform

**Comment by kcaponeAP** [comment]
Here is the detail in the XML that tells you what product to use. These are all possible combinations.

RequestType="LONGFORM" LoanRequestType="REVERSEMORTGAGE" 

RequestType="LONGFORM" LoanRequestType="REFINANCE" 
RequestType="SHORTFORM" LoanRequestType="REFINANCE" 

RequestType="SHORTFORM" LoanRequestType="PURCHASE" 
RequestType="LONGFORM" LoanRequestType="PURCHASE" 

RequestType="LONGFORM" LoanRequestType="PREQUALIFICATION" 
RequestType="LONGFORM" LoanRequestType="PERSONAL" 
RequestType="LONGFORM" LoanRequestType="HOMEEQUITY"

**Comment by cgrayson** [comment]
Setting `lendingtree.product` is part of the `unified` integration, and looks to be amply covered in that code. Does that field also need a value if the lead came in as a "shortform" format? Even though shortform leads haven't previously ever had it?

**Comment by alexkwolfe** [comment]
We used to only have one product (shortform) so there was really no need to have a product field. Now we have two products, it would make sense IMHO to populate that field for all leads that come from LendingTree.

**Comment by cgrayson** [comment]
Anything else, & does that change LGTY?

**Comment by alexkwolfe** [comment]
LGTM

**Pull request merged**